graph/path/internal/testgraphs/yen_shortest.go

```diff
@@ -0,0 +1,59 @@
+// Copyright ©2018 The Gonum Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+package testgraphs
+
+import (
+	"math"
+	
+	"gonum.org/v1/gonum/graph"
+	"gonum.org/v1/gonum/graph/simple"
+)
+
+func init() {
+	//for _, test := range YenShortestPathTests {
+		//if len(test.WantPaths) !=  {
+		//	panic(fmt.Sprintf("%q: bad shortest path test: non-unique paths marked unique", test.Name))
+		//}
+	//}
+}
+
+// ShortestPathTests are graphs used to test the static shortest path routines in path: BellmanFord,
+// DijkstraAllPaths, DijkstraFrom, FloydWarshall and Johnson, and the static degenerate case for the
+// dynamic shortest path routine in path/dynamic: DStarLite.
+var YenShortestPathTests = []struct {
+	Name              string
+	Graph             func() graph.WeightedEdgeAdder
+	Edges             []simple.WeightedEdge
+
+	Query         simple.Edge
+	K int
+	WantPaths     [][]int64
+}{
+	// Positive weighted graphs.
+	{
```

Comment:
I'd like to see tests that have k > |paths|, tests on a G={V={0,1}, E={(0,1)}}, on a G={V={0}, E=∅}, on a G=star_n, and on a G={V={-1,1,2..n}, E={(-1,m),(m,1)}} for m∈[2,n] and s=-1, t=1. for k<|{m}|, k = |{m}| and k>|{m}|.

That should probably cover it unless you have some other test cases that are well known, which would be good to have if they exist.